File: archive/c/c/file-input-output.c
#include <stdio.h>
#include <stdlib.h>

#define FILENAME "output.txt"

int write_file()
{
    FILE *handle;

    handle = fopen(FILENAME, "w");
    if (!handle) {
        perror("open file for writing");
        return 1;
    }

    fputs("A line of text\n", handle);
    fputs("Another line of text\n", handle);

    fclose(handle);
    return 0;
}

int read_file()
{
    char *line = NULL;
    ssize_t read;
    size_t len = 0;
    FILE *handle;

    handle = fopen(FILENAME, "r");
    if (!handle) {
        perror("open file for reading");
        return 1;
    }

    while ((read = getline(&line, &len, handle)) != -1) {
        printf("%s", line);
    }

    fclose(handle);
    free(line);

    return 0;
}

int main(int argc, char **argv)
{
    int rc;

    rc = write_file();
    if (rc != 0) {
        return EXIT_FAILURE;
    }

    rc = read_file();
    if (rc != 0) {
        return EXIT_FAILURE;
    }

    return EXIT_SUCCESS;
}

File: archive/r/rust/file-input-output.rs
use std::fs::File;
use std::io::prelude::*;
use std::io::Result;

fn write_file() -> Result<()> {
    let mut out = File::create("output.txt")?;

    let bytes_written = out.write(b"A line of text written to this file in Rust!\n").unwrap_or(0);
    let bytes_written2 = out.write(b"Another line also written by a Rust program!\n").unwrap_or(0);

    if bytes_written == 0 || bytes_written2 == 0 {
        ()
    }

    out.flush()?;

    Ok(())
}

fn read_file() -> Result<()> {
    let mut in_file = File::open("output.txt")?;

    let mut line = String::new();
    match in_file.read_to_string(&mut line) {
        Ok(_)   =>  println!("{}", line),
        Err(_)  =>  panic!("Could not read contents into string!")
    }

    Ok(())
}

fn main() -> Result<()> {
    write_file()?;
    Ok(read_file()?)
}